src/Microsoft.DotNet.Darc/Darc/Operations/VirtualMonoRepo/VmrDiffOperation.cs

```diff
@@ -0,0 +1,343 @@
+                    ?? Path.GetFileName(tmpProductRepo);
+                tmpVmrProductRepo = await PrepareVmr(repo2, tmpPath, mapping);
+            }
+            
+            await AddRemoteAndGenerateDiff(tmpProductRepo, tmpVmrProductRepo, repo2.Branch, await GetDiffFilters(mapping));
+        }
+        finally
+        {
+            if (fileSystem.DirectoryExists(tmpPath))
+            {
+                var gitFiles = Directory.GetDirectories(tmpPath, GitDirectory, SearchOption.AllDirectories)
+                    .Select(gitDir => Path.Combine(gitDir, "objects"))
+                    .SelectMany(q => Directory.GetFiles(q, "*", SearchOption.AllDirectories));
+                foreach (var gitFile in gitFiles)
+                {
+                    var fileInfo = new FileInfo(gitFile);
+                    fileInfo.Attributes = FileAttributes.Normal;
+                    fileInfo.IsReadOnly = false;
+                }
+                fileSystem.DeleteDirectory(tmpPath, true);
+            }
+        }
+        return 0;
+    }
+
+    private async Task<string> PrepareVmr(DiffRepo vmr, string tmpPath, string mapping)
+    {
+        if (string.IsNullOrEmpty(mapping))
+        {
+            throw new ArgumentException($"When preparing VMR, mapping can't be null");
+        }
+
+        var vmrProductRepo = Path.Combine(tmpPath, Guid.NewGuid().ToString());
+        if (vmr.IsLocal)
+        {
+            await CheckoutBranch(vmr);
+            CopyDirectory(Path.Combine(vmr.Path, VmrInfo.SourceDirName, mapping), vmrProductRepo, true);
+        }
+        else
+        {
+            vmrProductRepo = await PartiallyCloneVmr(tmpPath, vmr, mapping);
+        }
+        await GitInitRepo(vmrProductRepo, vmr.Branch);
+
+        return vmrProductRepo;
+    }
+
+    private async Task<string> PrepareRepo(DiffRepo repo, string tmpPath)
+    {
+        string tmpProductRepo = Path.Combine(tmpPath, Path.GetFileName(repo.Path));
+
+        if (!repo.IsLocal)
+        {
+            await processManager.ExecuteGit(Path.GetDirectoryName(tmpProductRepo)!, [
+                "clone",
+                "--depth", "1",
+                repo.Path,
+                "-b", repo.Branch,
+                tmpProductRepo
+            ]);
+        }
+        else
+        {
+            await CheckoutBranch(repo);
+            CopyDirectory(repo.Path, tmpProductRepo, true);
+        }
+
+        return tmpProductRepo;
+    }
+
+    private async Task<(DiffRepo repo1, DiffRepo repo2)> ParseInput()
+    {
+        DiffRepo repo1, repo2;
+        var parts = options.Input.Split("..", StringSplitOptions.RemoveEmptyEntries);
+        if (parts.Length > 2 || parts.Length < 1)
+        {
+            throw new ArgumentException($"Invalid input {options.Input}");
+        }
+        
+        if (parts.Length == 1)
+        {
+            var currentPath = Directory.GetCurrentDirectory();
```

Comment:
Please check what we do elsewhere - we have a method `FindGitRoot` or similar (I think on `IProcessManager`) so that you don't have to be in the root of the repo to call this

Reply:
done